### PR TITLE
Add Rc2 release changes

### PR DESCRIPTION
**Pull request by PaulHigin**
This PR contains changes for RC2 release, that include support for secret additional metadata.
Tests and documentation have been updated.

### CODE REVIEW

File: test/Microsoft.PowerShell.SecretStore.Tests.ps1 [hunk 1]
```diff
@@ -164,6 +164,72 @@ Describe "Test Microsoft.PowerShell.SecretStore module" -tags CI {
...
+        It "Verifies writing metadata along with secret content" {
+            $errorMsg = ""
+            $success = [Microsoft.PowerShell.SecretStore.LocalSecretStore]::GetInstance().WriteObject(
+                $secretName,
+                $secretContent,
+                (@{ TargetName = 'SomeName' }),
+                [ref] $errorMsg)
+
+            $success | Should -BeTrue
+            $errorMsg | Should -Be ""
```
Comment (anamnavi) [inline test/Microsoft.PowerShell.SecretStore.Tests.ps1:181]: you could do alternatively do ` $errorMsg | Should -BeNullOrEmpty` :)

**Review by anamnavi — APPROVED** [review]
LGTM

File: Docs/DesignDoc.md [hunk 2]
```diff
@@ -106,7 +106,10 @@ The data file structure consists of three main sections:
 The file data hash is a cryptographic hash of the key, iv, metadata, data, and is used to verify file content integrity.  
 
 The metadata is a json structure that contains information about each secret.
-It includes secret name, type, optional attributes, and encrypted blob size and offset information.  
+It includes secret name, type, optional attributes, and encrypted blob size and offset information.
```
Comment (SteveL-MSFT) [inline Docs/DesignDoc.md:109]: You don't have to make the change as part of this PR, but the convention seems to be an ARCHITECTURE.md file https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html

**Review by SteveL-MSFT — APPROVED** [review]
LGTM, one non-blocking comment

**Pull request merged**